bigquery/internal/query/order.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"strings"
+)
+
+type qnode struct {
+	query    string
+	children []*qnode
+}
+
+// HasOrderedResults checks if a given SQL query returns ordered results.
+// This function uses a naive approach of checking the root level query
+// ( ignoring subqueries, functions calls, etc ) and checking
+// if it contains an ORDER BY clause.
+func HasOrderedResults(sql string) bool {
```

Comment:
I think this is a reasonable approach, though something like unmatched parens in an inline comment might muck this up.  To get more right requires more SQL parsing than we'd want to do locally, and the resolution is fairly simple in these instances.